ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/bundle/BundlePluginBase.java

```diff
@@ -183,7 +183,7 @@ private void removeLibrary(IPluginLibrary library, IManifestHeader header) {
 		if (index >= 0) {
 			// copy up to the removed library
 			StringBuilder buffer = new StringBuilder();
-			for (int i = 0; i > index; i++) {
+			for (int i = 0; i < index; i++) {
 				buffer.append(value.charAt(i));
 			}
```

Comment:
What about replacing the entire loop just by the following?
```suggestion
			buffer.append(value, 0, index);
```
This would make the code simpler and maybe even faster.

The same could be done for the last loop that copies the remaining part.
It could just be `buffer.append(value, after, value.length());`.

Besides that I wonder if this code works correctly if the the library to remove is the last element. It looks like a trailing comma will then remain.
Or if the library to remove is the only element. Then the headers value will be empty, which is also an error.

Reply:
ok, lets use StringBuilder.append(CharSequence, start, end) also in the loop below. I don't feel confident to propose a fix for the other odds.